ui/src/components/MainScreen.tsx

```diff
@@ -11,50 +12,34 @@ import InvestorWallet from './InvestorWallet'
 
 import { WalletIcon } from '../icons/Icons'
 
-// May eventually want to switch to a proper routing
-// library for different pages in the app
-export enum Mode {
-  ROLE_SELECT_VIEW,
-  INVESTOR_VIEW,
-  INVESTOR_WALLET_VIEW
-}
-
 type Props = {
   onLogout: () => void;
 }
 
 /**
  * React component for the main screen of the `App`.
  */
-const MainScreen: React.FC<Props> = ({onLogout}) => {
-  const [ mode, setMode ] = useState(Mode.ROLE_SELECT_VIEW);
-
-  switch(mode) {
-    case Mode.ROLE_SELECT_VIEW:
-      return <RoleSelectScreen
-                setView={view => setMode(view)}
-                onLogout={onLogout}/>
-
-    case Mode.INVESTOR_VIEW:
-      return (
-        <Page view={mode} setView={view => setMode(view)} onLogout={onLogout}>
-          <WelcomeHeader/>
-        </Page>
-      )
-
-    case Mode.INVESTOR_WALLET_VIEW:
-      return (
-        <Page
-          view={mode}
-          menuTitle={<><WalletIcon/>Wallet</>}
-          setView={view => setMode(view)} onLogout={onLogout}
-        >
-          <InvestorWallet/>
-        </Page>
-      )
-    default:
-      return <></>
-  }
-};
+const MainScreen: React.FC<Props> = ({onLogout}) => (
+  <Switch>
+    <Route path='/role'>
+      <RoleSelectScreen onLogout={onLogout}/>
+    </Route>
+
+    <Route path='/wallet'>
+      <Page
+        menuTitle={<><WalletIcon/>Wallet</>}
+        onLogout={onLogout}
+      >
+        <InvestorWallet/>
+      </Page>
+    </Route>
+
+    <Route path='/'>
+      <Page onLogout={onLogout}>
+        <WelcomeHeader/>
+      </Page>
+    </Route>
```

Comment:
There is a specific "welcome header" for each role, so does this really ever get executed? Or, is it a few url segments deeper?